### PR TITLE
MM-47359 : avoid passing global state in post_actions.ts to addPostToSearchResults

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
avoid passing global state in post_actions.ts to addPostToSearchResults

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/21250
Jira: https://mattermost.atlassian.net/browse/MM-47359
#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note
avoid passing global state in post_actions.ts to addPostToSearchResults```


[MM-47359]: https://mattermost.atlassian.net/browse/MM-47359?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]

**Comment by nevyangelova** [comment]
@AbhinavVihan thank you, could you please rebase your fork with our master and fix the linting error?

File: actions/post_actions.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export function flagPost(postId: string) {
         const rhsState = getRhsState(state);
 
         if (rhsState === RHSStates.FLAG) {
-            addPostToSearchResults(postId, state, dispatch);
+            dispatch(addPostToSearchResults(postId) as Action);
```
Comment (M-ZubairAhmed) [inline actions/post_actions.ts:79]: Why do we need to typecast ?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: actions/post_actions.ts (hunk 1 above)
Comment (AbhinavVihan) [reply actions/post_actions.ts:79]: without typecasting, typescript is inferring its type as Thunk
***Type '(dispatch: DispatchFunc, getState: GetStateFunc) => void' is not assignable to type 'Thunk'.***

Comment (M-ZubairAhmed) [reply actions/post_actions.ts:79]: yes because the action is not typed correctly

File: actions/post_actions.ts [hunk 2]
```diff
@@ -155,7 +155,8 @@ export function searchForTerm(term: string) {
     };
 }
 
-function addPostToSearchResults(postId: string, state: GlobalState, dispatch: DispatchFunc) {
+const addPostToSearchResults = (postId: string) => (dispatch: DispatchFunc, getState: GetStateFunc) => {
+    const state = getState();
```
Comment (M-ZubairAhmed) [inline actions/post_actions.ts:159]:
```suggestion
function addPostToSearchResults(postId: string) {
    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
    .
    .
    .
    .
    
    return {data: true};
    };
  }
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by nevyangelova** [comment]
@AbhinavVihan please fix the linting error :)

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @AbhinavVihan 
Tested, looks good to merge.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/283850-773b33f-onprem-ent-webapp-pr-11814-mm-ee-test:773b33f/mochawesome.html) - no PR related failures.

**Comment by jgilliam17** [comment]
@nevyangelova I changed the ticket link to this PR per your comments https://mattermost.atlassian.net/browse/MM-47442
Let me know if this is right? Thanks

**Comment by nevyangelova** [comment]
amazing, thank you @jgilliam17

**Pull request merged**